packages/client/hmi-client/src/components/widgets/tera-toggleable-input.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+	<div v-if="isEditing" v-bind="$attrs" class="flex align-items-center gap-1">
+		<tera-input-text
+			ref="inputRef"
+			:model-value="modelValue"
+			@update:model-value="emit('update:model-value', $event)"
+			@keydown="handleKeyDown"
+		/>
+		<Button text icon="pi pi-times" @click="onCancel" />
+		<Button text icon="pi pi-check" @click="onConfirm" />
+	</div>
+	<Button v-else :class="$attrs.class" class="text-to-edit" text @click="onEdit">
+		<component class="btn-content" :is="tag">{{ modelValue }}</component>
+		<i class="pi pi-pencil" />
+	</Button>
+</template>
+
+<script setup lang="ts">
+import { ref, nextTick } from 'vue';
+import Button from 'primevue/button';
+import TeraInputText from '@/components/widgets/tera-input-text.vue';
+
+const props = withDefaults(
+	defineProps<{
+		modelValue: string;
+		tag: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
+	}>(),
+	{
+		tag: 'span'
+	}
+);
+
+const emit = defineEmits(['update:model-value']);
+
+let initialValue = '';
+
+const inputRef = ref<InstanceType<typeof TeraInputText> | null>(null);
```

Comment:
Is `typeof TeraInputText` just `Component`?

Reply:
`Component` is just a more generic version. `typeof TeraInputText` considers its properties, etc